packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWFInstanceManager.m

```diff
@@ -0,0 +1,100 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTWFInstanceManager.h"
+
+@interface FLTThreadSafeMapTable<KeyType, ObjectType> : NSObject
++ (nonnull FLTThreadSafeMapTable *)strongToStrongMapTable;
+- (void)setObject:(nonnull ObjectType)object forKey:(nonnull KeyType)key;
+- (void)removeObjectForKey:(nonnull KeyType)key;
+- (nullable ObjectType)objectForKey:(nonnull KeyType)key;
+@end
+
+@implementation FLTThreadSafeMapTable {
+  NSMapTable<id, id> *_table;
+  dispatch_queue_t _lockQueue;
+}
+
++ (FLTThreadSafeMapTable *)strongToStrongMapTable {
+  return [[FLTThreadSafeMapTable alloc] initWithTable:[NSMapTable strongToStrongObjectsMapTable]];
+}
+
+- (nonnull instancetype)initWithTable:(NSMapTable *)table {
+  self = [super init];
+  if (self) {
+    _lockQueue = dispatch_queue_create("FLTThreadSafeMapTable", DISPATCH_QUEUE_SERIAL);
+    _table = table;
+  }
+  return self;
+}
+
+- (void)setObject:(nonnull id)object forKey:(nonnull id)key {
+  if (key && object) {
```

Comment:
Why are you doing runtime checks for violations of the signature, but then also failing totally silently? That seems like the worst of both worlds.

I would either `assert`, or remove this entirely.